### PR TITLE
Implement is_one and Inv for Complex.

### PR DESCRIPTION
**Pull request by clarcharr**
Requires a new version of `num-traits` to be pushed.

### CODE REVIEW

**Comment by cuviper** [comment]
This should be good with a bump to num-traits 0.2.1.

File: src/lib.rs [hunk 1]
```diff
@@ -664,6 +664,24 @@ impl<'a, T: Clone + Num + Neg<Output = T>> Neg for &'a Complex<T> {
     }
 }
 
+impl<T: Clone + Num + Inv<Output = T>> Inv for Complex<T> {
```
Comment (cuviper) [inline src/lib.rs]: I don't think we need `T: Inv`, but we do need `T: Neg` for the inherent `fn inv`.

File: src/lib.rs [hunk 2]
```diff
@@ -664,6 +664,24 @@ impl<'a, T: Clone + Num + Neg<Output = T>> Neg for &'a Complex<T> {
...
+    type Output = Complex<T>;
+
+    #[inline]
+    fn inv(self) -> Complex<T> {
+        self.inv()
```
Comment (cuviper) [inline src/lib.rs]: I'm not sure, but you might have to make this explicitly `(&self).inv()` to avoid recursion.

**Comment by clarcharr** [comment]
Should be working now

**Comment by clarcharr** [comment]
I repeat, *now* it should be working.

**Comment by cuviper** [comment]
Thanks!

bors r+

**Pull request merged**